src/components/Tables/Table.tsx

```diff
@@ -89,41 +149,109 @@ const Table: React.FC<Props> = (props) => {
         }
     }, [selectCheckBoxes, selectAllCheckBox]);
 
-    // Fixes hydration error by re-rendering when the DOM is ready - https://nextjs.org/docs/messages/react-hydration-error
-    const [domLoaded, setDomLoaded] = useState(false);
-    useEffect(() => {
-        setDomLoaded(true);
-    }, []);
-
-    const columns: TableColumn<Row>[] = Object.entries(props.headers).map(
-        ([headerKey, headerName]) => {
-            return {
-                name: headerName,
-                selector: (row: Row) => row[headerKey],
-                sortable: true,
-            };
+    const columns: TableColumn<Row>[] = (
+        props.toggleableHeaders ? shownHeaders : props.headerKeysAndLabels
+    ).map(([headerKey, headerName]) => {
+        return {
+            name: headerName,
+            selector: (row) => row.data[headerKey],
+            minWidth: "12rem",
+            maxWidth: "20rem",
+            sortable: props.sortable ?? true,
```

Comment:
I think I'd prefer setting a default value in props as much as possible, so you can easily inspect what the default is (rather than defaulting to `undefined` and setting _somewhere_ in code!)